### ISSUE TITLE
ARM version

### ISSUE THREAD

**Issue by Serentty** [otya128/winevdm#1038]
I run Windows on an AArch64-based device (M1 Mac). Since this is already emulating a 16-bit x86 CPU, it would be nice to be able to run this on ARM Windows without having to go through two layers of emulation.

**Comment by ShadowEO** [comment]
I don't believe this would be possible. You also are misunderstanding how this works, WineVDM isn't "emulating" anything, it's doing syscall translations. You would still need a i386/amd64 processor emulator to run WineVDM, and to translate the 16-bit i386/AMD64 calls up to 64-bit.

**Comment by Anixx** [comment]
Wine relies on Linux kernel module that emulates 16-bit CPU.

**Comment by DarkShadow44** [comment]
> Wine relies on Linux kernel module that emulates 16-bit CPU.

Wine doesn't use emulation, it runs the program in 16bit protected mode.

It would be possible to make an ARM version, similar to Hangover for Wine, but it would probably be a lot of work.

**Comment by orospakr** [comment]
Hey, so wanted to chime in here: I just tried winevdm 0.9.0 this morning to run Encarta 95 on Windows 11 ARM. Presumably winevdm/otvdm itself was running within the x86 32/64 emulation feature of ARM Windows (itself running in Parallels on Apple Silicon, for anyone interested).

It works virtually perfectly. I'm pretty blown away. Here's a screenshot.

<img width="1321" alt="Screenshot 2023-11-03 at 11 55 14" src="https://github.com/otya128/winevdm/assets/16714/3968daca-dd5b-44c2-be60-a7b9fa701702">

**Comment by Anixx** [comment]
> Wine doesn't use emulation, it runs the program in 16bit protected mode.

I think you cannot activate a 16-bit protected mode under a 64-bit OS.

**Comment by cracyc** [comment]
> one issue, video playback doesn't work, missing codec).

You probably need to enable the indeo codecs which are included with windows 11 but disabled by default.

> I think you cannot activate a 16-bit protected mode under a 64-bit OS.

You can. Only v86 mode was disabled in long mode (at least until x86s which will remove 16bit protected mode entirely).  Microsoft disallows ldt access which is required but it still works in linux.

**Comment by haveamission** [comment]
@orospakr Yeah I was able to do the same thing as you - Windows 11 ARM (running in Parallels) 16-bit support worked. A bit unstable (application I tested froze briefly a couple times and then crashed, over maybe 10 minutes of usage), but for an alpha release, on hardware that is theoretically unsupported? Pretty good.

**Comment by nmssoft** [comment]
@orospakr  @haveamission  Did y'all get the installation to work.  I can manually drag the win16 exe onto otvdm or otvdmw.  But even replacing amd64 vs arm64 in the infs didn't work.  I can run like wintrek or simcity on it.

**Comment by kristibektashi** [comment]
> [@orospakr](https://github.com/orospakr) [@haveamission](https://github.com/haveamission) Did y'all get the installation to work. I can manually drag the win16 exe onto otvdm or otvdmw. But even replacing amd64 vs arm64 in the infs didn't work. I can run like wintrek or simcity on it.

Yes this is old, but for anyone who stumbles upon this issue, this can be fixed by installing this: https://github.com/otya128/ntvdm64.dll. Just download the latest release and follow the instructions. Sidenote: Windows 11 24H2 and later remove the mentioned "C:\Windows\SysArm32" folder, so if you're on those versions, just skip that part
